### ISSUE TITLE
Cannot read property 'error' of undefined

### ISSUE THREAD

**Issue by sunlubo** [Glavin001/atom-beautify#1196]
I'm trying to beautify my markdown but I get the error saying "Cannot read property 'error' of undefined"

'atom' version: 1.10.2
'atom-beautify' version: 0.29.12

Error Info: 
Cannot read property 'error' of undefined
TypeError: Cannot read property 'error' of undefined
  at /Users/sun/.atom/packages/atom-beautify/src/beautifiers/remark.coffee:42:10
  at Promise._execute (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/debuggability.js:299:9)
  at Promise._resolveFromExecutor (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:481:18)
  at new Promise (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:77:14)
  at Remark.module.exports.Remark.beautify (/Users/sun/.atom/packages/atom-beautify/src/beautifiers/remark.coffee:36:16)
  at /Users/sun/.atom/packages/atom-beautify/src/beautifiers/index.coffee:316:24
  at Promise._execute (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/debuggability.js:299:9)
  at Promise._resolveFromExecutor (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:481:18)
  at new Promise (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:77:14)
  at /Users/sun/.atom/packages/atom-beautify/src/beautifiers/index.coffee:238:18
  at tryCatcher (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/util.js:16:23)
  at Promise._settlePromiseFromHandler (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:510:31)
  at Promise._settlePromise (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:567:18)
  at Promise._settlePromise0 (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:612:10)
  at Promise._settlePromises (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:691:18)
  at Promise._fulfill (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:636:18)
  at PromiseArray._resolve (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise_array.js:125:19)
  at PromiseArray._promiseFulfilled (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise_array.js:143:14)
  at Promise._settlePromise (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:572:26)
  at Promise._settlePromise0 (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:612:10)
  at Promise._settlePromises (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:691:18)
  at Promise._fulfill (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:636:18)
  at Promise._resolveCallback (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:431:57)
  at Promise._settlePromiseFromHandler (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:522:17)
  at Promise._settlePromise (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:567:18)
  at Promise._settlePromise0 (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:612:10)
  at Promise._settlePromises (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:691:18)
  at Promise._fulfill (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:636:18)
  at Promise._resolveCallback (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:431:57)
  at Promise._settlePromiseFromHandler (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:522:17)
  at Promise._settlePromise (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:567:18)
  at Promise._settlePromise0 (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:612:10)
  at Promise._settlePromises (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:691:18)
  at Promise._fulfill (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:636:18)
  at Promise._resolveCallback (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:431:57)
  at ReductionPromiseArray._resolve (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/reduce.js:61:19)
  at Promise.completed [as _fulfillmentHandler0] (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/reduce.js:122:15)
  at Promise._settlePromise (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:564:21)
  at Promise._settlePromise0 (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:612:10)
  at Promise._settlePromises (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:691:18)
  at Async._drainQueue (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/async.js:138:16)
  at Async._drainQueues (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/async.js:148:10)
  at Async.drainQueues (/Users/sun/.atom/packages/atom-beautify/node_modules/bluebird/js/release/async.js:17:14)

**Comment by Terminl3nt** [comment]
yes I also found this bug, when  I save my markdown file.  I use the Atom Material  Theme

**Comment by mikewesthad** [comment]
Ditto - this bug only happens for me when using Remark. Tidy Markdown still works (though I find Tidy's formatting to be hard to read).

I had to install the last version `apm install atom-beautify@0.29.11` to get Remark working again.

**Comment by vmuriart** [comment]
Same problem on both .12 and .13, reverting to 0.29.11 did the trick

**Comment by garystafford** [comment]
I concur, both 0.29.12 and 0.29.13 threw the following error with Markdown, using Remark. Reverting to 0.29.11 fixed the issue. Thank you @mikewesthad and @vmuriart!

```text
Cannot read property 'error' of undefined
Hide Stack Trace
TypeError: Cannot read property 'error' of undefined
    at /Users/gstaffo/.atom/packages/atom-beautify/src/beautifiers/remark.coffee:42:10
    at Promise._execute (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/debuggability.js:299:9)
    at Promise._resolveFromExecutor (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:481:18)
    at new Promise (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:77:14)
    at Remark.module.exports.Remark.beautify (/Users/gstaffo/.atom/packages/atom-beautify/src/beautifiers/remark.coffee:36:16)
    at /Users/gstaffo/.atom/packages/atom-beautify/src/beautifiers/index.coffee:316:24
    at Promise._execute (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/debuggability.js:299:9)
    at Promise._resolveFromExecutor (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:481:18)
    at new Promise (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:77:14)
    at /Users/gstaffo/.atom/packages/atom-beautify/src/beautifiers/index.coffee:238:18
    at tryCatcher (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:510:31)
    at Promise._settlePromise (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:567:18)
    at Promise._settlePromise0 (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:612:10)
    at Promise._settlePromises (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:691:18)
    at Promise._fulfill (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:636:18)
    at PromiseArray._resolve (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise_array.js:125:19)
    at PromiseArray._promiseFulfilled (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise_array.js:143:14)
    at Promise._settlePromise (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:572:26)
    at Promise._settlePromise0 (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:612:10)
    at Promise._settlePromises (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:691:18)
    at Promise._fulfill (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:636:18)
    at Promise._resolveCallback (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:431:57)
    at Promise._settlePromiseFromHandler (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:522:17)
    at Promise._settlePromise (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:567:18)
    at Promise._settlePromise0 (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:612:10)
    at Promise._settlePromises (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:691:18)
    at Promise._fulfill (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:636:18)
    at Promise._resolveCallback (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:431:57)
    at Promise._settlePromiseFromHandler (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:522:17)
    at Promise._settlePromise (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:567:18)
    at Promise._settlePromise0 (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:612:10)
    at Promise._settlePromises (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:691:18)
    at Promise._fulfill (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:636:18)
    at Promise._resolveCallback (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:431:57)
    at ReductionPromiseArray._resolve (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/reduce.js:61:19)
    at Promise.completed [as _fulfillmentHandler0] (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/reduce.js:122:15)
    at Promise._settlePromise (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:564:21)
    at Promise._settlePromise0 (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:612:10)
    at Promise._settlePromises (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/promise.js:691:18)
    at Async._drainQueue (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/async.js:138:16)
    at Async._drainQueues (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/async.js:148:10)
    at Async.drainQueues (/Users/gstaffo/.atom/packages/atom-beautify/node_modules/bluebird/js/release/async.js:17:14)
```

**Comment by Glavin001** [comment]
Hmm this may be because of https://github.com/Glavin001/atom-beautify/commit/ba0988068e302da9be4e3734c5a282df989818f9

If someone submits a Pull Request reverting to a version of Remark that they confirm works I will merge and publish. Thank you!

**Comment by bemeurer** [comment]
Experiencing the same issue here using Atom 1.10.2 and atom-beautify 0.29.13

**Comment by Glavin001** [comment]
Published to v0.29.15
